chaos_genius/utils/datetime_helper.py

```diff
@@ -16,13 +18,55 @@ def get_rca_date_from_string(date_value):
     return datetime.strptime(date_value, "%Y/%m/%d %H:%M:%S").date()
 
 
-def get_date_string_with_tz(date_value) -> str:
+def get_datetime_string_with_tz(date_value, hourly=False) -> str:
     """Get date string with timezone."""
     tz_offset = SUPPORTED_TIMEZONES[TIMEZONE]
-    main_str = date_value.strftime("%d %b %Y") + f" {TIMEZONE}"
+    if hourly:
+        main_str = date_value.strftime("%d %b %Y %H:%M") + f" {TIMEZONE}"
+    else:
+        main_str = date_value.strftime("%d %b %Y") + f" {TIMEZONE}"
     return f"{main_str} ({tz_offset})" if tz_offset != "" else main_str
 
 
+def _get_tz_from_offset_str(utc_offset_str):
+    # TODO: update code when tz implementation is complete
+    sign = -1 if utc_offset_str[-6] == "-" else 1
+    utc_offset_mins = int(utc_offset_str[-2:]) * sign
+    utc_offset_hrs = int(utc_offset_str[-5:-3]) * sign
+
+    utc_offset = timedelta(hours=utc_offset_hrs, minutes=utc_offset_mins)
+
+    timezones = pytz.all_timezones
+    for tz_name in timezones:
+        try:
+            tz = pytz.timezone(tz_name)
+            tz_offset = tz._transition_info[-1][0]
```

Comment:
Can you link to a documentation for this field? It's a private field and the implementation of it can change anytime.

Could we use this function instead - https://github.com/stub42/pytz/blob/2ed682a7c4079042f50975970fc4f503c8450058/src/pytz/tzinfo.py#L396?

Reply:
Currently we can't. I'm planning to update the logic once we have more robust tz implementation.